app/scripts/decisions.directive.js

```diff
@@ -35,24 +32,86 @@ angular.module('dashboard')
+                    self.types = types;
                 }
                 else {
                     $log.error("dbDecisions.setTypes: bad args");
                 }
             }
 
+            function resetData() {
+                self.record = [];
+                self.supporter = [];
+                self.voting = [];
+            }
+
+            function updateDecision(aDecision) {
+                if (angular.isObject(aDecision)) {
+                    $log.log("dbDecisions.updateDecision", arguments);
+
+                    if (aDecision.typeName === CONST.MTGEVENT.MINUTEUPDATED) {
+                        if (angular.isObject(mtgItemSelected) && angular.equals(mtgItemSelected.meetingGuid, aDecision.meetingID)) {
+                            if (angular.isObject(mtgTopicSelected) && angular.equals(mtgTopicSelected.topicGuid, aDecision.topicGuid)) {
+
+                                if (angular.isObject(aDecision.minuteEntry)) {
+                                    if (!angular.isArray(self.record)) {
+                                        self.record = [];
+                                    }
+                                    if (!angular.isArray(self.supporter)) {
+                                        self.supporter = [];
+                                    }
+                                    var updated = false;
+                                    for (var index = 0; !updated && index < self.record.length; index++) {
+                                        var entry = self.record[index];
+                                        if (angular.isObject(entry) && angular.equals(entry.minuteEntryGuid, aDecision.minuteEntry.minuteEntryGuid)) {
+                                            angular.merge(entry, aDecision.minuteEntry);
+                                            updated = true;
+                                        }
+                                    }
+                                    if (!updated) {
+                                        self.record.push(aDecision.minuteEntry);
+                                    }
+                                }
+                                if (angular.isArray(aDecision.supporters) && aDecision.supporters.length) {
+                                    // todo: implement supporter updating
+                                }
+                            }
+                        }
+                    }
+                    else if (aDecision.typeName === CONST.MTGEVENT.MINUTEDELETED) {
+                        if (angular.isObject(mtgItemSelected) && angular.equals(mtgItemSelected.meetingGuid, aDecision.meetingID)) {
+                            if (angular.isObject(mtgTopicSelected) && angular.equals(mtgTopicSelected.topicGuid, aDecision.topicGuid)) {
+
+                                var search = angular.isArray(self.record);
+                                for (var i = self.record.length + CONST.NOTFOUND; search && i > CONST.NOTFOUND; i--) {
```

Comment:
A matter of taste, to me `length + CONST.NOTFOUND` looks more cryptic in this context than `length - 1`
Naturally this won't block the PR :)